cmd/cosign/cli/sign/sign.go

```diff
@@ -530,16 +530,18 @@ func keylessSigner(ctx context.Context, ko options.KeyOpts) (*SignerVerifier, er
 }
 
 func SignerFromKeyOpts(ctx context.Context, certPath string, certChainPath string, ko options.KeyOpts) (*SignerVerifier, error) {
```

Comment:
IMO this is clearer if we have `fulcio.NewSigner` take in a SignerVerifier. Then, the flow can be something like:

```go
var sv *SignerVerifier
genKey := false
switch {
case ko.Sk:
    sv = signerFromSecurityKey(ko.Slot)
case ko.KeyRef != "":
    sv = signerFromKeyRef(...)
default:
    genKey = true
    fmt.Fprintln(os.Stderr, "Generating ephemeral keys....")
    sv = signerFromNewKey() // <-- new function
}

if ko.IssueCertificate || genKey {
    sv = keylessSigner(...)
}
return sv
```